### PR TITLE
Remove hardcoded catid's in tests

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Does remove the hardcoded categories in the system tests. Additionally it does some restructure in the db tasks file to be more logic.

Is a preparation pr for #40733.

### CODE REVIEW

File: tests/System/integration/administrator/components/com_banners/Banner.cy.js [hunk 1]
```diff
@@ -12,16 +16,16 @@ describe('Test in backend that the banners form', () => {
...
     cy.clickToolbarButton('Cancel');
 
     cy.wait('@listview');
   });
 
   it('can edit a banner', () => {
-    cy.db_createBanner({ name: 'Test Banner' }).then((id) => {
-      cy.visit(`administrator/index.php?option=com_banners&task=banner.edit&id=${id}`);
+    cy.db_createBanner({ name: 'Test Banner' }).then((banner) => {
+      cy.visit(`administrator/index.php?option=com_banners&task=banner.edit&id=${banner.id}`);
```
Comment (sandewt) [inline tests/System/integration/administrator/components/com_banners/Banner.cy.js]: forward slash ?

File: tests/System/integration/administrator/components/com_contact/Contact.cy.js [hunk 2]
```diff
@@ -12,8 +16,8 @@ describe('Test in backend that the contact form', () => {
   });
 
   it('can change access level of a test contact', () => {
-    cy.db_createContact({ name: 'Test contact' }).then((id) => {
-      cy.visit(`administrator/index.php?option=com_contact&task=contact.edit&id=${id}`);
+    cy.db_createContact({ name: 'Test contact' }).then((contact) => {
+      cy.visit(`administrator/index.php?option=com_contact&task=contact.edit&id=${contact.id}`);
```
Comment (sandewt) [inline tests/System/integration/administrator/components/com_contact/Contact.cy.js]: forward slash ?

**Comment by heelc29** [comment]
@laoneo Could you please add the milestone for J4.4?

**Comment by laoneo** [comment]
Thanks for the reminder

**Pull request merged**